modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupDXPCloudForm/index.jsx

```diff
@@ -187,6 +190,100 @@ const SetupDXPCloudPage = ({
 			setFormAlreadySubmitted(true);
 		}
 
+		const handleDataSubmit = async () => {
+			const {data: addDXPCloudEnvironmentResponse} = await client.mutate({
+				context: {
+					displaySuccess: false,
+					type: 'liferay-rest',
+				},
+				mutation: addDXPCloudEnvironment,
+				variables: {
+					DXPCloudEnvironment: {
+						accountKey: project.accountKey,
+						dataCenterRegion: dxp.dataCenterRegion,
+						disasterDataCenterRegion: dxp.disasterDataCenterRegion,
+						projectId: dxp.projectId,
+						r_accountEntryToDXPCloudEnvironment_accountEntryId:
+							project?.id,
+					},
+				},
+			});
+
+			if (addDXPCloudEnvironmentResponse) {
+				const dxpCloudEnvironmentId =
+					addDXPCloudEnvironmentResponse?.createDXPCloudEnvironment
+						?.id;
+
+				await Promise.all(
+					dxp.admins.map(({email, firstName, github, lastName}) =>
+						client.mutate({
+							context: {
+								displaySuccess: false,
+								type: 'liferay-rest',
+							},
+							mutation: addAdminDXPCloud,
+							variables: {
+								AdminDXPCloud: {
+									dxpCloudEnvironmentId,
+									emailAddress: email,
+									firstName,
+									githubUsername: github,
+									lastName,
+									r_accountEntryToAdminDXPCloud_accountEntryId:
+										project?.id,
+								},
+							},
+						})
+					)
+				);
+			}
+
+			await client.mutate({
+				context: {
+					type: 'liferay-rest',
+				},
+				mutation: updateAccountSubscriptionGroups,
+				variables: {
+					accountSubscriptionGroup: {
+						accountKey: project.accountKey,
+						activationStatus: STATUS_TAG_TYPE_NAMES.inProgress,
+						r_accountEntryToAccountSubscriptionGroup_accountEntryId:
+							project.id,
+					},
+					id: subscriptionGroupId,
+				},
+			});
+
+			if (featureFlags.includes('LPS-187767')) {
+				const notificationTemplateService = new NotificationQueueService(
+					client
+				);
+
+				try {
+					await notificationTemplateService.send(
```

Comment:
I am not completely sure everything that is happening here, but feels odd to be sending notifications/emails from the front end. What is the use case for getting into this try block? and is it something we could/should do with workflow/object actions?

Reply:
It's an e-mail notification structured via the notification template if I'm not mistaken, but this structure already existed, I just put the whole process inside a function so that I could call the function too when I'm in the error condition due to conflict, so I kept this structure. 

We're already using object actions in this notification template too.